cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -279,12 +286,12 @@ public ApplicationDetail getAppDetail(ApplicationId appId) throws Exception {
    * @throws Exception if ensureAccess throws exception
    */
   public boolean updateAppAllowed(ApplicationId appId) throws Exception {
-    ensureAccess(appId);
     ApplicationSpecification appSpec = store.getApplication(appId);
     if (appSpec == null) {
       // App does not exist. Allow to create a new one
       return true;
     }
+    ensureAccess(appId);
```

Comment:
Sounds good. However, we can still do the access check before looking up from the store, right?